### DISCUSSION TITLE
Can I graph finite faults in Gempy, so that they don't cut off my entire strata series?

### DISCUSSION THREAD

**Discussion by shairethomg** [gempy-project/gempy#943] [Q&A] [answered]
I'm trying to make this fault only cut me up to the second stratum, and I've tried placing the series in the order of the succession of events but it doesn't work. I also tried to modify the correlation matrix. Can someone please tell me how to graph the finite faults?
thanks.
![image](https://github.com/user-attachments/assets/6403c442-c919-4d64-b322-015e93d8b256)
![image](https://github.com/user-attachments/assets/4fcad8e5-c767-415b-bc9e-cb008c8f33c7)
![image](https://github.com/user-attachments/assets/e58f15ab-4e80-41ac-a69c-853fc3e8ca80)

**Comment by javoha** [comment]
Hi,
thanks for your question. Just to make sure: You want the fault to only displace the _mioceno_medio_ element, right? _mioceno_superior_ and _plioceno_inferior_ are supposed to be unaffected?
Actually you did everything correct and the result is exactly what you want then. The only probelm is that gempy atm always displays the fautls over the whole thing. This is just a visualization artefact. In your model the top two units are not actually displaced by the fault.
There are ways to change the visualization using some pyvista magic I guess if that is really important to you.
Let me know if that helps or if I got the question wrong.
Cheers,
Jan

**Comment by shairethomg** [comment]
Hi, that's right, that's my question, thank you very much.
Do you know how you could use Pyvista to improve this?

**Reply by NilsChudalla** [reply, marked as answer]
Hi @shairethomg , 

if it comes to Pyvista usage, you might stumble upon boolean operations (Example: https://docs.pyvista.org/examples/01-filter/boolean-operations.html). Those might be not sufficient for you, as they require manifold/watertight geometries, which these surface are not. If it is a onetime time thing, you could export the scene of your pyvista plotter object with
<code>plotter = gpv.plot_3d(geo_model)
plotter.p.export_obj("name_of_scene.obj")</code>. 
and then remove the vertices in some modeling software such as Blender. If you have to do it multiple times, I cannot think of any other way than creating a manifold object (e.g. <code>pv.Box(model xmin, model xmax, model ymin, model ymax, depth of intersection, model zmax)</code>) and doing the boolean operation with that.

**Reply by shairethomg** [reply]
Thank you very much!
